src/sinks/datadog/logs/tests.rs

```diff
@@ -55,58 +103,26 @@ async fn smoke_json() {
     }
 }
 
-async fn smoke_test() -> (Vec<String>, Vec<(http::request::Parts, Bytes)>) {
-    let (expected, rx) = start_test(StatusCode::OK, BatchStatus::Delivered).await;
-
-    let output = rx.take(expected.len()).collect::<Vec<_>>().await;
-
-    (expected, output)
-}
-
 #[tokio::test]
-async fn handles_failure_json() {
-    handles_failure().await;
-}
-
+/// Assert delivery error behavior
+///
+/// In the event that delivery fails -- in this case becaues it is FORBIDDEN --
+/// there should be no outbound messages from the sink. That is, receiving from
+/// its Receiver must fail.
 async fn handles_failure() {
     let (_expected, mut rx) = start_test(StatusCode::FORBIDDEN, BatchStatus::Failed).await;
+    let res = rx.try_next();
 
-    assert!(matches!(rx.try_next(), Err(TryRecvError { .. })));
-}
-
-async fn start_test(
-    http_status: StatusCode,
-    batch_status: BatchStatus,
-) -> (Vec<String>, Receiver<(http::request::Parts, Bytes)>) {
-    let config = indoc! {r#"
-            default_api_key = "atoken"
-            compression = "none"
-            batch.max_events = 1
-        "#};
-    let (mut config, cx) = load_sink::<DatadogLogsConfig>(&config).unwrap();
-
-    let addr = next_addr();
-    // Swap out the endpoint so we can force send it
-    // to our local server
-    let endpoint = format!("http://{}", addr);
-    config.endpoint = Some(endpoint.clone());
-
-    let (sink, _) = config.build(cx).await.unwrap();
-
-    let (rx, _trigger, server) = build_test_server_status(addr, http_status);
-    tokio::spawn(server);
-
-    let (batch, mut receiver) = BatchNotifier::new_with_receiver();
-    let (expected, events) = random_lines_with_stream(100, 10, Some(batch));
-
-    let _ = sink.run(events).await.unwrap();
-
-    assert_eq!(receiver.try_recv(), Ok(batch_status));
-
-    (expected, rx)
+    assert!(matches!(res, Err(TryRecvError { .. })));
 }
 
 #[tokio::test]
+/// Assert that metadata API keys are passed correctly
+///
+/// Datadog sink payloads come with an associated API key. This key can be set
+/// per-event or set in the message for an entire payload. This test asserts
+/// that, for successful transmission, the API key set in metadata is
+/// propagated.
```

Comment:
Is it worth testing having different API keys in each event, to ensure they are propagated independently?

Reply:
I believe that's what `multiple_api_keys` below does.